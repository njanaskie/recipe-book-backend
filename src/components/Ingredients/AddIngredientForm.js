import React, { useState, useContext } from 'react'
import { useSelector, useDispatch } from 'react-redux'
import IngredientsContext from '../../../context/ingredients-context'
import database from '../../firebase/firebase'


const AddIngredientForm = () => {
    const [name, setName] = useState('')
    const [category, setCategory] = useState('')
    const [price, setPrice] = useState('')
    const { dispatch } = useContext(IngredientsContext)

    const addIngredient = (e) => {
        e.preventDefault()
        const ingredient = {
            name,
            category,
            price
        }

        database.ref('ingredients').push(ingredient).then((ref) => {
            dispatch(({ type: 'ADD_INGREDIENT', ingredient: {id: ref.key, ...ingredient} }))
        })

        setName('')
        setCategory('')
        setPrice('')
    }

    return (
        <div>
            <p>Add Ingredient</p>
            <form onSubmit={addIngredient}>
                <input
                    type='text'
                    placeholder='Name'
                    autoFocus
                    value={name}
                    onChange={(e) => setName(e.target.value)}
                />
                <input
                    type='text'
                    placeholder='Category'
                    autoFocus
                    value={category}
                    onChange={(e) => setCategory(e.target.value)}
                />
                <input
                    type='text'
                    placeholder='Price'
                    autoFocus
                    value={price}
                    onChange={(e) => setPrice(e.target.value)}
                />
                <button>Add Ingredient</button>
            </form>
        </div>

    )
}

export default AddIngredientForm