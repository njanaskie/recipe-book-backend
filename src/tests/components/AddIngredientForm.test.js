import React from 'react'
import { shallow } from 'enzyme'
import AddIngredientForm from '../../components/Ingredients/AddIngredientForm'

test('should render AddIngredientForm correctly',() => {
    const wrapper = shallow(<AddIngredientForm />)
    expect(wrapper).toMatchSnapshot()
})