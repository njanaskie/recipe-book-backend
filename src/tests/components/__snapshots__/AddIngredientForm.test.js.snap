// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AddIngredientForm correctly 1`] = `
<div>
  <p>
    Add Ingredient
  </p>
  <form>
    <input
      placeholder="Name"
      type="text"
    />
    <input
      placeholder="Category"
      type="text"
    />
    <input
      placeholder="Price"
      type="text"
    />
  </form>
</div>
`;
